src/mainClass.js

```diff
@@ -0,0 +1,26 @@
+import { getData, getCategories } from "./source";
+
+export class AllItems {
+  constructor(searchCategory, displayCategoryLink) {
+    this.allCategories = searchCategory; // searchCategory()
+    this.linksUpdate = displayCategoryLink; // displayCategoryLink()
+    this.categories = [];
+  }
+
+  displayCategories() {
+    getCategories().then((res) => {
+      this.categories = res;
+      this.allCategories();
+    });
+  }
+
+  displayLinks(category) {
+    getData(category).then((res) => {
+     this.linksUpdate(res, category);
```

Comment:
keep links in property also.
You forgot to implement search links as part of task. So add search state for categories and links in properti, implement 2 more methods to UI consumes filtered arrays and draw from this class.